medusa.py

```diff
@@ -1596,6 +1598,15 @@ def my_message_handler(self, message, payload) -> None:
             if "trscrpt|" in data:
                 result = self.translator.translate(data[data.index("trscrpt|") + len("trscrpt|"):])
                 self.script.post({"my_data": result})
+            elif "tlskeylog|" in data:
```

Comment:
In order to avoid hardcoding the module in run_frida calls, why we don't handle the key storing within the handler ? 
for example If we modify the module as follows:

```js
  var packageName = null;
    var currentTime = null;
    var bufferedKeys = []; // Buffer to hold keys until the app is initialized

    // Function to log buffered keys
    function logBufferedKeys() {
        bufferedKeys.forEach(function(key) {
            send(packageName + '|' + currentTime + '|' + 'tlskeylog|' + key);
        });
        bufferedKeys = []; // Clear the buffer after logging
    }

    // Function to setup native hooks
    function setupNativeHooks() {
        try {
            Module.ensureInitialized('libssl.so');
        } catch (err) {
            console.log('libssl.so module not loaded. Trying to manually load it.');
            Module.load('libssl.so');
        }

        var SSL_CTX_set_keylog_callback = new NativeFunction(
            Module.findExportByName('libssl.so', 'SSL_CTX_set_keylog_callback'),
            'void',
            ['pointer', 'pointer']
        );

        var keylog_callback = new NativeCallback(function (ctxPtr, linePtr) {
            var keyLine = linePtr.readCString();
            // Check if packageName and currentTime are already set
            if (packageName && currentTime) {
                console.log(packageName + '|' + currentTime + '|' + 'tlskeylog|' + keyLine);
            } else {
                // Buffer the key for later logging
                bufferedKeys.push(keyLine);
            }
        }, 'void', ['pointer', 'pointer']);

        Interceptor.attach(Module.findExportByName('libssl.so', 'SSL_new'), {
            onEnter: function (args) {
                SSL_CTX_set_keylog_callback(args[0], keylog_callback);
            }
        });
    }

    // Function to attempt to get application info
    function tryGetAppInfo() {
        var app = Java.use("android.app.ActivityThread").currentApplication();
        if (app !== null) {
            var context = app.getApplicationContext();
            if (context !== null && packageName === null && currentTime === null) {
                packageName = context.getPackageName();
                currentTime = Math.floor(Date.now() / 1000);
                
                // Log any buffered keys now that packageName and currentTime are set
                logBufferedKeys();
            } else if (packageName === null || currentTime === null) {
                console.log("Waiting for application context...");
                setTimeout(tryGetAppInfo, 1000); // Retry after 1 second
            }
        } else {
            console.log("Application is not initialized yet, retrying...");
            setTimeout(tryGetAppInfo, 1000); // Retry after 1 second
        }
    }

    // Setup native hooks immediately to start capturing keys
    setupNativeHooks();

    // Attempt to get application info and log buffered keys when ready
    tryGetAppInfo();
````

The script will send the following data to the handler:

````
com.foo.bar | 1709204436 | tlskeylog | CLIENT_HANDSHAKE_TRAFFIC_SECRET a04b7f85def9764a8642a54af3878cf7241ff9486a3d7c9047b0bfbdbec64561 b15f87f93708dea08d01e43d3189f2afd840f6a2bfefaddabaad44d092383e69

com.foo.bar | 1709204436 | tlskeylog | SERVER_HANDSHAKE_TRAFFIC_SECRET a04b7f85def9764a8642a54af3878cf7241ff9486a3d7c9047b0bfbdbec64561 1235efeb1a286da9eb7063cc77bf8762887ed91e95a9efa9585196e61655542d

com.foo.bar| 1709204436 | tlskeylog | CLIENT_HANDSHAKE_TRAFFIC_SECRET 99b4bf597ef1aff5e325fe76942da9d79cd0eb6b6aabd6b20cb5fc4a4106bc8a 7aa0a2fe274b184dee9ef4964b71753a6d6e5e20353fd5c546f2579b82da7032

com.foo.bar | 1709204436 | tlskeylog | SERVER_HANDSHAKE_TRAFFIC_SECRET 99b4bf597ef1aff5e325fe76942da9d79cd0eb6b6aabd6b20cb5fc4a4106bc8a a7aba6083632ee1ea114b2c3514d05e225adfe7df124ad3b729e1f28aff5a761
...
...
````
So the tlskeylog carries the info regarding which file it should be stored to. So, when the handler receives the data, we can take the package name, unixtime (this is the same for all the tlskeylog during a single session), construct the file name and finally store the data. 

What do you think ?